Fix images expiring, allow reels/videos & save permalink in database

SvenWesterlaken opened the pull request:
A few items were missing or were causing errors in my production environment. These were:

- Reels did not show up, we wanted to use the thumbnails of these in the feed
- Permalinks (links to the actual post on Instagram) were missing
- Most importantly, the image links expire after 1 week as these were not updated if a post already existed in the database

These commits should fix these issues. Furthermore, It is improved with extra data now being saved in the database that might be useful for broader use cases.

kepes commented:
Hi, thanks for the PR. I have questions:

- I don't understand the comment for public links. This links will be on a public page, I don't understand you concern about security risk.
- A Reels and Video media not included for reason. The quality of this pictures often much worst than pictures so I don't want to put them on my sites. If you would like to include it please add a checkbox to the Settings hash and to the Admin pages to control this feature. For enable or disable and please put disable by default due currently the Video not downloaded and I don't want to change the plugin's behavior.

SvenWesterlaken commented:
ah, I understand. Adding an extra checkbox or settings in the plugin to enable/disable the feature, sounds like a good idea. I will look into that and add these changes to this PR.

For the public links, I don't really understand which one you refer to as I did not mention anything about security issues. The only things I changed related to links are:
- `originalURL` is now replaced every time with a new fetch as these links were invalid after a week.
- `permalink` was added as I needed a link to the actual Instagram post itself (which is saved in that property)

kepes commented:
Sorry, my mistake I mixed it up with another PR where you change README. Somehow this PR also reffering to that one. Never mind.

What is important is the checkbox on the interface. Thank you very much if you do!

SvenWesterlaken commented:
Added the checkbox as mentioned @kepes 
![image](https://user-images.githubusercontent.com/16725310/232618517-76bc1d2c-b93c-4170-a424-99784a600582.png)

SvenWesterlaken commented:
Any updates on merging this pull request @kepes?

zadinvit commented on server/content-types/image/schema.json:
```diff
@@ -32,18 +32,27 @@
       "type": "datetime"
     },
     "caption": {
-      "type": "text",
```
Hi, @SvenWesterlaken you missing comma in this row. It's causing error while run.

SvenWesterlaken replied:
Fixed in my last commit, thanks for mentioning @zadinvit

azrulaham2020 commented:
Hi any updates on this? We are also using this plugin and without this plugin we had no option other than discarding it? Hope we can have a fix for this issue soonest

pchurro commented:
@kepes Would highly appreciate merging this pull request. Thank you (and @SvenWesterlaken )

SvenWesterlaken commented:
I am considering making a fork if this is not merged by February (I'm currently too busy with client projects). I will link it here once I get it up and running as a Strapi package.

EDIT: make that march. Got more work than planned in the end (as almost every dev will understand)

SvenWesterlaken commented:
Better late than never: https://github.com/SvenWesterlaken/strapi-plugin-instagram-images

Here should be the package with the same base including the fixes proposed in this PR, but maintained by me from now on. @pchurro @azrulaham2020 

Note that the URL changed to `/instagram-images/` from `/instagram/` to prevent conflicts.

pchurro commented:
SO appreciated @SvenWesterlaken 🙌🙏

zadinvit commented:
@SvenWesterlaken Maybe add some migration info for people which used this version. 

I found that now you must:

1.  add info to plugins config
2. end point changed from /instagram/images to /instagram-images/images

SvenWesterlaken commented:
> @SvenWesterlaken Maybe add some migration info for people which used this version.
> 
> I found that now you must:
> 
> 1. add info to plugins config
> 2. end point changed from /instagram/images to /instagram-images/images

Planning on adding some parts to the README so will take this with those changes
